plasmapy/diagnostics/tests/test_langmuir.py

```diff
@@ -1,40 +1,86 @@
-"""Tests for functions that use Langmuir analysis functions."""
+# coding=utf-8
+"""Tests for Langmuir probe analysis functions."""
 
 import numpy as np
+from astropy import units as u
+import pytest
+import plasmapy.constants as const
 
-from plasmapy.diagnostics.langmuir import (swept_probe_analysis,
-                                           obtain_EEDF,
-                                           obtain_EEPF)
-
-
-class Test_swept_probe_analysis():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Performing a swept probe analysis should return an error "
-                  "when zero signal is passed, but doesn't.")
-        assert True, errStr
-
-
-class Test_obtain_EEDF():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Attempting to obtain the EEDF should return an error when "
-                  "zero signal is passed, but doesn't.")
-        assert True, errStr
-
-
-class Test_obtain_EEPF():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Attempting to obtain the EEPF should return an error when "
-                  "zero signal is passed, but doesn't.")
-        assert True, errStr
+from plasmapy.diagnostics.langmuir import (Characteristic,
+                                           swept_probe_analysis)
+
+N = 30  # length of test characteristic
+
+I_arr = np.random.rand(N) * u.A
+U_arr = np.random.rand(N) * u.V
+class Test__characteristic_errors:
+    r"""Test the Characteristic class constructor in langmuir.py"""
+    N = 30  # length of test characteristic
+    U_infarr = np.append(np.random.rand(N - 1), np.inf) * u.V
+    I_infarr = np.append(np.random.rand(N - 1), np.inf) * u.A
+
+    def test_infinite_I(self):
```

Comment:
methods should have docstrings describing what is tested.

Reply:
Gee, I think we might be too strict here :laughing: When these tests are simple properly written like here, the function name suffices most of the time. This function raises ValueError. Why? Because there's something wrong in the arguments. What? The infinite current. Simple!

Reply:
I guess I'm being strict here because sometimes the intent behind a test should be obvious, but it isn't (whether it's due to how it was initially written or edits further down the line). So it's nice to have the intent behind the test explicitly stated. Like here, it could literally be one line.